packages/integration-tests/index.js

```diff
@@ -308,41 +394,26 @@ function addCustomUrlPathToCollectionFiles(collection, customFilePath) {
  * @param {string} [postfix] - string to append to collection name
  * @param {string} [customFilePath]
  * @param {string} [duplicateHandling]
- * @returns {Promise.<number>} number of collections added
+ * @returns {Promise<number>} number of collections added
  */
 async function addCollections(stackName, bucketName, dataDirectory, postfix,
   customFilePath, duplicateHandling) {
-  const collections = await setupSeedData(stackName, bucketName, dataDirectory);
-  const promises = collections.map((collection) => limit(() => {
-    if (postfix) {
-      collection.name += postfix;
-      collection.dataType += postfix;
-    }
-    if (customFilePath) {
-      collection.files = addCustomUrlPathToCollectionFiles(collection, customFilePath);
-    }
-    if (duplicateHandling) {
-      collection.duplicateHandling = duplicateHandling;
-    }
-    const c = new Collection();
-    const id = constructCollectionId(collection.name, collection.version);
-    console.log(`Adding collection ${id}`);
-    return c.delete(collection)
-      .then(() => api.addCollectionApi({ prefix: stackName, collection }));
-  }));
-  return Promise.all(promises).then((cs) => cs.length);
-}
+  const rawCollections = await readJsonFilesFromDir(dataDirectory);
+
+  const collections = rawCollections.map(
+    (collection) => buildCollection({
+      collection,
+      customFilePath,
+      duplicateHandling,
+      postfix
+    })
+  );
 
-/**
- * Return a list of collections
- *
- * @param {string} stackName - CloudFormation stack name
- * @param {string} bucketName - S3 internal bucket name
- * @param {string} dataDirectory - the directory of collection json files
- * @returns {Promise.<Array>} list of collections
- */
-async function listCollections(stackName, bucketName, dataDirectory) {
-  return setupSeedData(stackName, bucketName, dataDirectory);
+  await Promise.all(
+    collections.map((collection) => addCollection(stackName, collection))
+  );
+
+  return rawCollections.length;
```

Comment:
If any of the `addCollection` calls in the above `Promise.all` failed, then you would only have added some amount of collections which is less than `rawCollections.length`?

Reply:
If any of them fail, it's going to throw an exception and never return a length, which was the case before this change as well.